File: Sample/MaaSAdvertising-iOS-Sample/MasterViewController.m
//
//  MasterViewController.m
//  PWAdvertisingQA
//
//  Created on 4/15/16.
//  Copyright © 2016 Phunware, Inc. All rights reserved.
//

#import "MasterViewController.h"
#import "DetailViewController.h"
#import "AdUnit.h"

@interface MasterViewController ()

@property NSMutableArray *objects;

@end

@implementation MasterViewController {
    NSArray * _adObjects;
}

- (void)viewDidLoad {
    [super viewDidLoad];
    
    // Do any additional setup after loading the view, typically from a nib.
    _adObjects = [AdUnit ads];
    self.detailViewController = (DetailViewController *)[[self.splitViewController.viewControllers lastObject] topViewController];
}

- (void)viewWillAppear:(BOOL)animated {
    self.clearsSelectionOnViewWillAppear = self.splitViewController.isCollapsed;
    [super viewWillAppear:animated];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


#pragma mark - Segues

- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
    if ([[segue identifier] isEqualToString:@"showDetail"]) {
        NSIndexPath *indexPath = [self.tableView indexPathForSelectedRow];
        DetailViewController *controller = (DetailViewController *)[[segue destinationViewController] topViewController];
        controller.adObject = _adObjects[indexPath.row];
        controller.navigationItem.leftBarButtonItem = self.splitViewController.displayModeButtonItem;
        controller.navigationItem.leftItemsSupplementBackButton = YES;
    }
}

#pragma mark - Table View

- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView {
    return 1;
}

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
    return _adObjects.count;
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:@"Cell" forIndexPath:indexPath];
    AdUnit *adObject = _adObjects[indexPath.row];
    cell.textLabel.text = adObject.adName;
    return cell;
}

- (BOOL)tableView:(UITableView *)tableView canEditRowAtIndexPath:(NSIndexPath *)indexPath {
    return NO;
}


@end

File: Samples/Sample/Sample/MasterViewController.swift
//
//  MasterViewController.swift
//  Sample
//
//  Created on 8/10/17.
//  Copyright © 2017 Phunware, Inc. All rights reserved.
//

import UIKit

enum AdType: Int {
    case banner
    case interstitial
    case video
    case rewardedVideo
    case native
    case landingPage
    
    func description() -> String {
        switch self {
        case .banner:
            return "Banner"
        case .interstitial:
            return "Interstitial"
        case .video:
            return "Video"
        case .rewardedVideo:
            return "Rewarded Video"
        case .native:
            return "Native"
        case .landingPage:
            return "Landing Page"
        }
    }
}

class MasterViewController: UITableViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
    }

    // MARK: - Navigation

    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
        if let detailVC = segue.destination as? DetailViewController,
            segue.identifier == "Detail",
            let selectedRow = tableView.indexPathForSelectedRow?.row,
            let selectedType = AdType(rawValue: selectedRow) {
            detailVC.adType = selectedType
        }
    }
}

extension MasterViewController {
    
    // MARK: - Table view data source
    
    override func numberOfSections(in tableView: UITableView) -> Int {
        return 1
    }
    
    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return 6
    }
    
    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        let cell = tableView.dequeueReusableCell(withIdentifier: "Cell", for: indexPath)
        
        let type = AdType(rawValue: indexPath.row)!
        cell.textLabel?.text = type.description()
        
        return cell
    }
}
